api/api/accounts/serializers.py

```diff
@@ -0,0 +1,48 @@
+from rest_framework import serializers
+
+from .models import Faculty, Staff, Student, User
+
+
+class StudentSerializer(serializers.ModelSerializer):
+    class Meta:
+        model = Student
+        fields = ['roll_no', 'batch', 'department', 'hostel_address', 'bio']
+
+
+class FacultySerializer(serializers.ModelSerializer):
+    class Meta:
+        model = Faculty
+        fields = ['title', 'department', 'designation']
+
+
+class StaffSerializer(serializers.ModelSerializer):
+    class Meta:
+        model = Staff
+        fields = ['department', 'designation']
+
+
+class UserSerializer(serializers.ModelSerializer):
+    student = StudentSerializer(many=False, allow_null=True, required=False)
+    faculty = FacultySerializer(many=False, allow_null=True, required=False)
+    staff = StaffSerializer(many=False, allow_null=True, required=False)
+
+    class Meta:
+        model = User
+        fields = ['id', 'email', 'first_name', 'last_name', 'gender',
```

Comment:
id should be a read only field. Or it can be removed as the end user doesn't create the id.